### PR TITLE
Comp4 and Comp6

### PR DESCRIPTION
**Pull request by sdinesh-90**

### CODE REVIEW

File: PSICompile/ILCodeGen.cs [hunk 1]
```diff
@@ -121,21 +161,31 @@ public class ILCodeGen : Visitor {
...
-      else {
-         string op = b.Op.Kind.ToString ().ToLower ();
-         op = op switch { "mod" => "rem", "eq" => "ceq", "lt" => "clt", _ => op };
-         Out ($"    {op}");
-      }
+      else  Out ($"    {OpCode (b.Op.Kind)}");
    }
-   
+
+   string OpCode (Token.E opKind) {
```
Comment (tarydon) [inline PSICompile/ILCodeGen.cs:166]: Good.

File: PSICompile/ILCodeGen.cs [hunk 2]
```diff
@@ -24,6 +24,9 @@ public class ILCodeGen : Visitor {
       Out ("    ret");
       Out ("  }");
       Out ("}");
+      // Finally add the labels
+      foreach (var (idx, label) in mInserts.OrderByDescending (a => a.Idx))
+         S.Insert (idx, label);
```
Comment (tarydon) [inline PSICompile/ILCodeGen.cs]: Why do this? Not clear to me.

Comment (sdinesh-90) [reply PSICompile/ILCodeGen.cs]: Just saw how while statement is implemented. Will do the same for if statement.

**Comment by sdinesh-90** [comment]
While reviewing other code for Comp4 and Comp6, I found out that IFElse statement is not working as we need two variables. Need to do more work.

**Pull request closed without merging**